### ISSUE TITLE
Add configuration option to hide Window Decorations

### ISSUE THREAD

**Issue by joshuataylor** [labwc/labwc#751]
tldr: I prefer to not show Window Decorations at all.

It would be great to add a config option that would allow users to hide window decorations, something like this:
```xml
<core>
  <decoration>server</decoration>
  <show_decorations>false</show_decorations>
</core>
```

Server side decorations would need to be enabled for this to work, I think?

In [view.c](https://github.com/labwc/labwc/blob/master/src/view.c#L656), there is this code block:

```c
void
view_set_decorations(struct view *view, bool decorations)
{
	assert(view);
	if (view->ssd_enabled != decorations && !view->fullscreen) {
		/*
		 * Set view->ssd_enabled first since it is referenced
		 * within the call tree of ssd_create()
		 */
		view->ssd_enabled = decorations;
		if (decorations && rc.show_decorations != true) {
			decorate(view);
		} else {
			undecorate(view);
		}
		view_apply_special_geometry(view);
	}
}
```

Here we could check if the config option has been set, then only decorate if it hasn't. By default it will decorate.

Right now I just have that codeblock commented out and use that. :-)

I'd be happy to implement this feature if this is something that is in scope of labwc and others agree.

Ideally, the initial implementation would consist of the following:

1. Add a configuration option to core to toggle window decorations
2. Add the check to that function (I think that's the best spot for it?)

Would love to get others thoughts on this :-)

**Comment by Consolatis** [comment]
I don't see an issue with providing that option, we could potentially just add a new value for the existing `decorations` flag, e.g. something like 
```xml
<decorations>none</decorations>
```
We would need to explain in the docs that this setting may not prevent applications to still use client side decorations even though we told them that we prefer server side ones.

Regarding the correct place to check, that one could do it. I think we also have a check somewhere else which decides on the initial state of `view->ssd_enabled`, likely split into a xwayland and a xdg one. Would need to look into that.

**Comment by johanmalm** [comment]
I prefer to stay consistent with `Openbox 3.6` unless there is good reason to deviate.

Consider

```XML
<applications>
  <application class="*">
    <decor>no</decor>
  </application>
</applications>
```

With `s/class/app_id/` ?

Ref: http://openbox.org/wiki/Help:Applications

The following is also related:

```XML
<theme>
  <keepBorder>yes</keepBorder>
</theme>
```

We held off on implementing `<applications>` so far, but I think it feels right to go for it at some point.
See this comment for reference: https://github.com/labwc/labwc/pull/433#issuecomment-1383085004

**Comment by hype3** [comment]
Is there any timeframe when this will be implented? Will it also be possible to use thicker borders if I want?

**Comment by joshuataylor** [comment]
Hi!

Been super busy with some personal stuff, but this issue is just for tracking hiding the Window Decorations, so it goes from this:

![2023-02-20_16-43](https://user-images.githubusercontent.com/225131/220056110-983b1a68-e042-4200-85eb-c4dc6c79c675.png)

to this:
![2023-02-20_16-41](https://user-images.githubusercontent.com/225131/220056133-f4b4a058-4ad6-47dc-b5e5-5f80784ed844.png)

**Comment by Consolatis** [comment]
> Will it also be possible to use thicker borders if I want?

That should work already: you can create a file called `themerc-override` in `~/.config/labwc/` containing
`border.width: 20`

**Comment by hype3** [comment]
> > Will it also be possible to use thicker borders if I want?
> 
> That should be possible already: create a file called `themerc-override` in `~/.config/labwc/` containing `border.width: 20` and then do a `labwc --reconfigure` (or restart labwc)

Ok that I didn't know about thx.

**Comment by dglava** [comment]
>But when I hide window decorations also the border is hidden
No idea if labwc wants to stick to Openbox's behaviour, but I wanted to mention that that is also the behaviour in Openbox when hiding the decorations. I do agree that it looks prettier with the border (when undecorated), so ideally that would be an option inside the theme or labwc config?

**Comment by johanmalm** [comment]
> > But when I hide window decorations also the border is hidden
> 
> No idea if labwc wants to stick to Openbox's behaviour, but I wanted to mention that that is also the behaviour in Openbox when hiding the decorations. I do agree that it looks prettier with the border (when undecorated), so ideally that would be an option inside the theme or labwc config?

Controlled by the following in accordance with openbox 3.6 - but not yet implemented in labwc.

```
<theme>
  <keepBorder>yes</keepBorder>
</theme>
```

Agree. I like a border when the rest of SDD is hidden :smile:

**Comment by heroin-moose** [comment]
Yeah, would be nice to have.

**Comment by Consolatis** [comment]
This might be non-trivial to implement as we are currently only having 2 states: SSD enabled and SSD disabled.
As all calculated positioning (maximize, tiling) requires offsetting by the SSD margin that could get a bit tricky.

**Comment by johanmalm** [comment]
Hmm… yes. I haven’t looked at code, did think that it might need so refactoring.

**Comment by Consolatis** [comment]
Regarding the screenshots above, this seems more like the issue we currently have with the default settings for the SSD decorations for Qt applications. E.g.
- #594

**Comment by Consolatis** [comment]
This should now be possible with something like this:
```xml
<windowRules>
    <windowRule identifier="*" serverDecoration="no" />
</windowRules>
```
Please comment if it isn't working or there are issues with the new feature.
